_posts/2021-05-04-how-to-update-your-ref-arch.adoc

```diff
@@ -384,47 +384,72 @@ function fuzzy_import_move_state {
+. Roll out with `terragrunt apply`.
+
+Some of the services, such as EKS and ASG, have slightly different steps to the above list, so please pay attention to that.
+
+Now for the full-fledged instructions to upgrade a single service:
+
+[arabic]
+. Check the service's downstream dependencies. Use the `graph-dependencies` command to create
+  a visual representation. The arrow points from the leaf to the root, toward the dependency. Thus in the graph,
+  the top nodes are leaf nodes and the bottom, root nodes.
     * `terragrunt graph-dependencies | dot -Tpng > graph.png`
     * If you get an error that `dot` is not available, install https://www.graphviz.org/download/[graphviz], which
       installs the `dot` utility.
+    * Here is an example of a dependency tree for the `dev` account using Reference Architecture v1:
+
+image::/assets/img/guides/ref-arch-v2-upgrade/ref-arch-v2-dep-tree.png[Dependency Graph]
 
-. First, you need to ensure the module is updated to the same version used in Reference Architecture version
-  https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme/releases/tag/v0.0.1-20201125[20201125], or
-  newer.
+. Ensure the module is updated to the same version used in Reference Architecture version
+  https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme/releases/tag/v0.0.1-20201125[20201125].
+    * If you're running a newer version, continue.
     * If you are running an older version, follow the migration guides referenced in
       https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme/releases[the Reference Architecture
-      releases] to update to the latest version. This is necessary because the Service Catalog module references use newer
+      releases] to update to the latest version. This is important because the Service Catalog module references use newer
       versions from the Module Catalog than what is shipped with v1.0 of the Reference Architecture. Once you've upgraded to
       https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme/releases/tag/v0.0.1-20201125[20201125],
-      you'll be at a known starting point and can automate any state manipulations that are required to update a service.
+      you can automate any state manipulations that are required to update a service using the provided guides and scripts.
 
-. Make modifications to the `terragrunt.hcl` file to be compatible with the Service Catalog:
+. *Make a backup of the state file* using the following command: `terragrunt state pull > backup.tfstate`
+   You can use this to rollback the state to before you attempted the migration with the
+   following command: `terragrunt state push "$(pwd)/backup.tfstate"`.
+
+* NOTE: Make sure to use the Terraform version that is required for your module, as specified in the
+  https://www.terraform.io/docs/language/settings/index.html#specifying-a-required-terraform-version[required_version]
+  configuration of the module.
```

Comment:
@yorinasub17 Would like your input on this! I moved the state file backup step earlier, to before they update the terragrunt.hcl, because I have seen on my system that after making the changes to the config, the state pull doesn't behave. I think it's because the source points to the service catalog, rather than the old infra-modules. Should this not be the case?

If we're not 100% on this, I would rather err conservatively and tell users to backup before any changes.

Reply:
@yorinasub17 Bump in case missed